### ISSUE TITLE
Msys2 version of gdb 9.2-3 not compatible with mingw-w64-x86_64-gcc 10.2.0-6

### ISSUE THREAD

**Issue by gbburkhardt** [msys2/MSYS2-packages#2361]
## Describe the issue

Breakpoints set using Msys2 gdb 9.2-3 don't work with programs compiled with mingw-w64-x86_64-gcc 10.2.0-6

## Steps to Reproduce the Problem

Glenn.Burkhardt@AI234087 MSYS ~
$ g++ -g t.cpp -o t

$ gdb ./t
GNU gdb (GDB) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-msys".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Traceback (most recent call last):
  File "<string>", line 3, in <module>
ModuleNotFoundError: No module named 'libstdcxx'
/etc/gdbinit:6: Error in sourced command file:
Error while executing Python code.
Reading symbols from ./t...
(gdb) b main
Breakpoint 1 at 0x14000154d: file t.cpp, line 6.
(gdb) l
1       #define __USE_MINGW_ANSI_STDIO 1
2       #include <stdio.h>
3
4       int main()
5       {
6           size_t z = 0;
7
8           printf("%zu\n", z);
9       }
(gdb) b 8
Breakpoint 2 at 0x140001555: file t.cpp, line 8.
(gdb) r
Starting program: /home/glenn.burkhardt/t [New Thread 42340.0xaee8]
Warning:
Cannot insert breakpoint 1.
Cannot access memory at address 0x140001548 Cannot insert breakpoint 2.
Cannot access memory at address 0x140001555

Command aborted.

## Additional Context: Operating System, Screenshots

- OS: [e.g. Windows 10 Pro version 1809 64bit]

**Comment by Biswa96** [comment]
Did you try with [mingw-w64-x86_64-gdb][1] package?

[1]: https://packages.msys2.org/package/mingw-w64-x86_64-gdb?repo=mingw64

**Comment by gbburkhardt** [comment]
Yes.  That works.  But command line editing doesn't in an Msys2 shell.

**Comment by Biswa96** [comment]
msys2 and mingw shell should not be mixed. Use the mingw prefixed packages in mingw shell and non prefixed packages in msys2 shell.

**Comment by mati865** [comment]
> 
> Yes.  That works.  But command line editing doesn't in an Msys2 shell.

It works if you run it like this `winpty gdb`, the same thing applies to LLDB so you have to run it with `winpty lldb`.

**Comment by jeremyd2019** [comment]
the msys2 gdb probably just needs to be updated to support Windows ASLR.

**Comment by Biswa96** [comment]
That fork issue with cygwin?

**Comment by jeremyd2019** [comment]
no, https://sourceware.org/bugzilla/show_bug.cgi?id=26037

**Comment by gbburkhardt** [comment]
Thanks.  I never guessed.  Yes, command line editing works for 
mingw-w64-x86_64-gdb 10.1-2 with 'winpty'.  One might expect it to "just 
work".  Command line editing for mingw-w64-x86_64-gdb 10.1-2 in a shell 
invoked with 'msys2_shell.cmd -mingw64' doesn't work without 'winpty'.

I didn't expect 'gdb' to be a Windows console program.

**Comment by a7v7** [comment]
This issue has a surprising side effect. It can "break" debugging in Eclipse. When I set up a new computer, I only configured c:\msys64\bin into Win10 environment variables. This was to add msys2 make. Unfortunately, gdb is located in the same directory. To get eclipse to work correctly with msys2 in win10, you must put the compiler directory (c:\msys64\mingw64\bin) before the make (c:\msys64\usr\bin) directory like this:
![fixed_gdb](https://user-images.githubusercontent.com/46093926/112736097-65312b00-8f0d-11eb-8130-8906afb25589.png)

If you do not, the failure mode looks like this:
![broken_gdb](https://user-images.githubusercontent.com/46093926/112736196-11731180-8f0e-11eb-9935-8a3a5ecffb74.png)

The thing to note is that the fix is not obvious from the errors.
